lib/base-compiler.js

```diff
@@ -784,8 +821,33 @@ export class BaseCompiler {
         return result;
     }
 
-    getCacheKey(source, options, backendOptions, filters, tools, libraries) {
-        return {compiler: this.compiler, source, options, backendOptions, filters, tools, libraries};
+    getCacheKey(source, options, backendOptions, filters, tools, libraries, files) {
+        if (files) {
+            return {compiler: this.compiler, source, options, backendOptions, filters, tools, libraries, files};
```

Comment:
Doesn't this work just as well if `files` isn't set?

Reply:
Maybe? I kind of wanted to make sure we're not polluting the cache. Will need to test

Reply:
I'm not convinced, it's fine on stringify and if (key.files), but I don't know how LRU() will hash it as - any idea?